libgadget/gravshort-tree.c

```diff
@@ -118,14 +117,18 @@ grav_short_tree(const ActiveParticles * act, PetaPM * pm, ForceTree * tree, doub
     priv.NeutrinoTracer = NeutrinoTracer;
     priv.G = pm->G;
     priv.cbrtrho0 = pow(rho0, 1.0 / 3);
+    priv.Ti_Current = Ti_Current;
 
     if(!tree->moments_computed_flag)
         endrun(2, "Gravtree called before tree moments computed!\n");
 
     tw->ev_label = "GRAVTREE";
     tw->visit = (TreeWalkVisitFunction) force_treeev_shortrange;
-    /* gravity applies to all particles. Including Tracer particles to enhance numerical stability. */
-    tw->haswork = NULL;
+    /* gravity applies to all gravitationally active particles.
+     * If we are a PM step then all particles are active. */
+    tw->haswork = gravshort_haswork;
+    if(!act->ActiveParticle)
```

Comment:
This may be easier to read:

if(act->ActiveParticle) {
  tw->haswork = NULL;
} else {
  tw->haswork = gravshort_haswork;
}

Also, can we / shall we move the conditional into gravshort_haswork? (e.g. if !act_activeParticle return True) -- Because haswork function is supposed to be "the" conditional.

Reply:
Yes you are right! I will adjust the condition. We maybe should but cannot move it into gravshort_haswork because we make memory allocation decisions in treewalk_build_queue depending on whether ->haswork is NULL.